### PR TITLE
Update dependency management

### PR DESCRIPTION
**Pull request by bonniegee**
- Add dependabot config
- Add requirements.in file
- Update pre-commit config
- Add virtual environment directory to .gitignore

### CODE REVIEW

**Review by p-galligan — COMMENTED** [review]
I don't know a ton about the asterism version you're updating from, but you may want to check to see whether you require a new migration. 

Both @helrond and I had to make one when upgrading apps that use Asterism.

**Comment by helrond** [comment]
I'm not seeing Django asking for a migration so I think you're okay there.

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+asterism==0.7.2
+bagit==1.8.1
+bagit-profile==1.3.1
+colorlog==3.2.0
```
Comment (helrond) [inline requirements.in]: `colorlog` does not need to be in the top level requirements

File: requirements.in [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+Django==3.2.4
```
Comment (helrond) [inline requirements.in]: Django 3.2.5 has been released.

File: requirements.in [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+djangorestframework==3.11.2
```
Comment (helrond) [inline requirements.in]: DRF is at 3.12.4

**Review by helrond — CHANGES_REQUESTED** [review]
A couple of requirements to update.

**Comment by p-galligan** [comment]
Now that I've pulled it down and tried to stand it up, I realized I don't have a config since you're .gitignoring it.

Maybe add some info into the readme about the types of info that it should include?

**Comment by helrond** [comment]
Well, there's an example config file. The approach we've taken [in other cases](https://github.com/RockefellerArchiveCenter/gemini/blob/base/entrypoint.sh#L9-L11) is to check for that file in the entrypoint script and create it if it doesn't exist.

**Comment by p-galligan** [comment]
> Well, there's an example config file. The approach we've taken [in other cases](https://github.com/RockefellerArchiveCenter/gemini/blob/base/entrypoint.sh#L9-L11) is to check for that file in the entrypoint script and create it if it doesn't exist.

Good call. My head is all messed up today lol. Anyway, I'd vote for adding something like that to the entrypoint here.

**Review by helrond — APPROVED** [review]
Just converted `urlize_quoted_links` and `urlize`

**Review by p-galligan — APPROVED** [review]
Looks good now!

**Review by HaSistrunk — APPROVED** [review]

**Pull request merged**